example/spec/helpers/kinesisHelpers.js

```diff
@@ -146,30 +150,31 @@ async function putRecordOnStream(streamName, record) {
   }).promise();
 }
 
+
 /**
- *  Wait until an exectution matching the desired execution starts.
+ * Wait for test stepfunction execution to exist.
  *
  * @param {string} recordIdentifier - random string identifying correct execution for test
  * @param {integer} maxWaitTime - maximum time to wait for the correct execution in milliseconds
  * @param {string} firstStep - The name of the first step of the workflow, used to query if the workflow has started.
  * @returns {Object} - {executionArn: <arn>, status: <status>}
  * @throws {Error} - any AWS error, re-thrown from AWS execution or 'Workflow Never Started'.
  */
-async function waitForTestSfStarted(recordIdentifier, maxWaitTime, firstStep = 'SfSnsReport') {
+async function waitForTestSf(recordIdentifier, maxWaitTime, firstStep = 'SfSnsReport') {
   let timeWaited = 0;
-  let lastExecution;
   let workflowExecution;
 
   /* eslint-disable no-await-in-loop */
   while (timeWaited < maxWaitTime && workflowExecution === undefined) {
     await timeout(waitPeriodMs);
     timeWaited += waitPeriodMs;
-    lastExecution = await getLastExecution();
-    // getLastExecution returns undefined if no previous execution exists
-    if (lastExecution && lastExecution.executionArn) {
-      const taskInput = await lambdaStep.getStepInput(lastExecution.executionArn, firstStep);
+    const executions = await getExecutions();
+    // Search all recent executions for target recordIdentifier
+    for (const execution of executions) {
+      const taskInput = await lambdaStep.getStepInput(execution.executionArn, firstStep);
       if (taskInput !== null && taskInput.payload.identifier === recordIdentifier) {
-        workflowExecution = lastExecution;
+        workflowExecution = execution;
+        break;
       }
```

Comment:
Happy to see a more JS idiomatic  way to do this, but I was failing to come up with it.